### PR TITLE
feat: add transfer from recovered account; test recovery flow.

### PR DESCRIPTION
**Pull request by benoitdevos**
logion-network/logion-internal#460

### CODE REVIEW

File: packages/client/integration/ProtectionRecovery.spec.ts [hunk 1]
```diff
@@ -51,43 +68,95 @@ describe("Recovery SDK", () => {
...
 
         const accepted = (await pending.refresh()) as AcceptedProtection;
-        let activate = await accepted.activate(signer);
+        console.log("Activating")
+        return accepted.activate(signer);
+    }
 
-        expect(activate).toBeInstanceOf(ActiveProtection);
-    });
+    it("recovers a lost account", async () => {
```
Comment (gdethier) [inline packages/client/integration/ProtectionRecovery.spec.ts:80]: Splitting the test into 2 cases requires the cases to be executed in the right order. Also, one generally does not expect a dependency between separate cases. Therefore, I would merge the 2 cases but keep your refactoring to make the "big" test case more readable.

**Review by gdethier — CHANGES_REQUESTED** [review]
Good job! However, I think that the integration test has to be adapted. Let's discuss.

File: packages/client/integration/ProtectionRecovery.spec.ts (hunk 1 above)
Comment (benoitdevos) [reply packages/client/integration/ProtectionRecovery.spec.ts:80]: The 2 test cases are completely independent, and can be run in any order.
The test case `enable protection` runs using `REQUESTER_ADDRESS`.
The test case `recovers a lost account` runs using `LOST_ADDRESS` (and `NEW_ADDRESS`). For this latter case, a protection is enabled via the call `await enableProtection(LOST_ADDRESS)` at line 83.

**Review by gdethier — APPROVED** [review]
OK after explanation (the 2 cases are independent because there are now 3 requesters).

File: packages/client/integration/ProtectionRecovery.spec.ts (hunk 1 above)
Comment (benoitdevos) [reply packages/client/integration/ProtectionRecovery.spec.ts:80]: Yes.

**Pull request merged**